EpsilonLights/Core/Src/Lights.c

```diff
@@ -219,28 +214,37 @@ void reportLightsToCanTask(void const* arg)
             continue;
         }
 
+        if (HAL_CAN_GetTxMailboxesFreeLevel(&hcan) == 0)
+        {
+            osMutexRelease(canHandleMutex);
+            continue;
+        }
+
+
         // Toggle blue LED for every CAN message sent
         HAL_GPIO_TogglePin(LED_BLUE_GPIO_Port, LED_BLUE_Pin);
         // Set CAN msg address
-        hcan2.pTxMsg->StdId = LIGHTS_STATUS_STDID;
+        canTxHdr.StdId = LIGHTS_STATUS_STDID;
+        uint8_t data [1];
+        uint32_t mailbox;
         // Initalize to avoid garbage values in [6] anad [7]
         LightsStatus stat = {0};
         // Read lights gpios
-        stat.lowBeams = HAL_GPIO_ReadPin(HLOW_GPIO_Port, HLOW_Pin);
-        stat.highBeams = HAL_GPIO_ReadPin(HHIGH_GPIO_Port, HHIGH_Pin);
+        stat.lowBeams = HAL_GPIO_ReadPin(HEAD_GPIO_Port, HEAD_Pin);
+        stat.highBeams = 0;
```

Comment:
hmmmm, at some point the LightStatus should no longer be reporting any `highBeams` fields and just have a headlights one. New issue? Given the changes we need to make elsewhere.

Reply:
Yeah I'll make an issue for it in JIRA and then we can update this and driver controls